src/scss/init/unity-layout/_unity-layout.scss

```diff
@@ -0,0 +1,199 @@
+// is-number($value)
+//
+// Returns true if $value variable is of type number
+//
+// Weight: -7
+//
+// Style guide: init.libraries.unity-layout.is-number
+@function is-number($value) {
+  @return type_of($value) == 'number';
+}
+
+// is-integer($value)
+//
+// Returns true if $value variable is an integer
+//
+// Weight: -6
+//
+// Style guide: init.libraries.unity-layout.is-integer
+@function is-integer($value) {
+  @return is-number($value) and round($value) == $value;
+}
+
+// is-percentage($value)
+//
+// Return true if $value variable is a percentage with no decimals
+//
+// Weight: -5
+//
+// Style guide: init.libraries.unity-layout.is-percentage
+@function is-percentage($value) {
+  @return is-integer($value) and unit($value) == '%';
+}
+
+// convert-percentage($value)
+//
+// Check if $value variable is a valid integer or percentage value. If $value is an integer, but not a percentage
+// then return a percentage value. If $value is neither an integer, nor a percentage, then return false.
+//
+// Weight: -4
+//
+// Style guide: init.libraries.unity-layout.convert-percentage
+@function convert-percentage($value) {
+  @if is-integer($value) and not(is-percentage($value)) {
+    $percent-value: $value * 1%;
+    @return $percent-value;
+  } @else if is-percentage($value) {
+    @return $value;
+  } @else {
+    @return false;
+  }
+}
+
+// is-valid($values)
+//
+// Pass each $value in $values list through convert-percentage() function. Return the new list of $values if all values
+// are valid, if not, return false.
+//
+// Weight: -3
+//
+// Style guide: init.libraries.unity-layout.is-valid
+@function is-valid($values) {
+  $new-values: ();
+  @for $i from 1 through length($values) {
+    $value: nth($values, $i);
+    $new-value: convert-percentage($value);
+    $new-values: append($new-values, $new-value);
+
+    @if not($new-value) {
+      @return false;
+    }
+  }
+  @return $new-values;
+}
+
+
+// value-sum($values)
+//
+// Return true if the sum of $values is less than or equal to 100.
+//
+// Weight: -2
+//
+// Style guide: init.libraries.unity-layout.value-sum
+@function value-sum($values) {
+  $sum: 0;
+  @each $value in $values {
+    $sum: $sum + $value;
+  }
+  @return $sum <= 100;
+}
+
+// unity-layout-values-valid($values)
+//
+// Return true if all values are valid percentages and the sum of $values is less than or equal to 100.
+//
+// Weight: -1
+//
+// Style guide: init.libraries.unity-layout.unity-layout-values-valid
+@function unity-layout-values-valid($values) {
```

Comment:
make this a prviate scss function for this file, so declare it as _unity-layout-values-valid